src/teacher.py

```diff
@@ -75,6 +80,81 @@ def _answer_Demonstration_query(
                                           _train_task_idx=query.train_task_idx)
         return DemonstrationResponse(query, teacher_traj)
 
+    def _answer_StateBasedDemonstration_query(
+            self, state: State,
+            query: StateBasedDemonstrationQuery) -> DemonstrationResponse:
+        # The query is asking for a demonstration from the current state to
+        # the goal state. Planning to a low-level state is hard. This
+        # implementation currently assumes that only one option is required
+        # to get from the state to the goal state. Furthermore, the option
+        # is identified in an environment-specific way. This is all a stand-in
+        # for a human teacher. Note that although these trajectories are good,
+        # they are not demonstrations per se, because they do not reach task
+        # goals (and are not necessarily associated with any particular task).
+        trajectory = None
+        null_response = DemonstrationResponse(query, teacher_traj=None)
+        goal_state = query.goal_state
+        if CFG.env == "cover_multistep_options":
+            assert CFG.cover_multistep_use_learned_equivalents
+            # Setup.
+            block_type, target_type, robot_type = _get_types_by_names(
+                CFG.env, ["block", "target", "robot"])
+            Pick, Place = _get_options_by_names(
```

Comment:
so you're using the Learned Equivalents rather than the basic Pick/Place options? i'm not sure i fully understand the significance of using one versus the other...

Reply:
the learned equivalent ones have objects and parameter spaces that should exactly match those of the learned options. that's not true for Pick/Place, especially re: the parameter spaces.